src/planner/ompl/Planner.cpp

```diff
@@ -374,75 +376,78 @@ trajectory::InterpolatedPtr planCRRTConnect(
-{
-      auto ompl_state = sspace->allocState(_originalTraj->getWaypoint(idx));
-      path->append(ompl_state);
-      sspace->freeState(ompl_state);
-}
+  // Step 3: Use the OMPL methods to simplify the path
+  ::ompl::geometric::PathSimplifier simplifier = ::ompl::geometric::PathSimplifier(si);
 
-// Step 3: Use the OMPL methods to simplify the path
-::ompl::geometric::PathSimplifierPtr simplifier;
-simplifier.reset(new ::ompl::geometric::PathSimplifier(si));
+  // Flag for user to know if shorten was successful
+  bool shorten_success = false;
 
-// Set the parameters for termination of simplification process
-::ompl::time::point const time_before = ::ompl::time::now();
-::ompl::time::point time_current;
-::ompl::time::duration const time_limit = ::ompl::time::seconds(_timeout);
-double empty_steps = 0;
+  // Set the parameters for termination of simplification process
+  ::ompl::time::point const time_before = ::ompl::time::now();
+  ::ompl::time::point time_current;
+  ::ompl::time::duration const time_limit = ::ompl::time::seconds(_timeout);
+  double empty_steps = 0;
+  
+  do 
+  {
 
-do 
-{
+    bool const shortened = simplifier.shortcutPath(*path, 1, _maxEmptySteps, 1.0, 0.0);
+    empty_steps = (empty_steps + !shortened)*!shortened; // Increment only if shorten fails consecutively
+    time_current = ::ompl::time::now();
+    shorten_success = shorten_success || shortened; // TODO: There should be a better way to do this.
 
-  bool const shortened = simplifier->shortcutPath(*path, 1, _maxEmptySteps, _rangeRatio, _snapToVertex);
-  empty_steps += !shortened;
-  time_current = ::ompl::time::now();
+  } while(time_current - time_before <= time_limit && empty_steps <= _maxEmptySteps);
 
-} while(time_current - time_before <= time_limit && empty_steps <= _maxEmptySteps);
+  // Step 4: Convert the simplified geomteric path to AIKIDO untimed trajectory
+  auto returnTraj = dart::common::make_unique<trajectory::Interpolated>(
+        std::move(_stateSpace), std::move(_interpolator));
 
-// Step 4: Convert the simplified geomteric path to AIKIDO untimed trajectory
-auto returnTraj = std::make_shared<trajectory::Interpolated>(
-      std::move(_stateSpace), std::move(_interpolator));
+  for (size_t idx = 0; idx < path->getStateCount(); ++idx) 
+  {
+        const auto *st =
+            static_cast<GeometricStateSpace::StateType *>(
+                path->getState(idx));
+        // Arbitrary timing
+        returnTraj->addWaypoint(idx, st->mState);
+  }
 
-for (size_t idx = 0; idx < path->getStateCount(); ++idx) 
-{
-      const auto *st =
-          static_cast<GeometricStateSpace::StateType *>(
-              path->getState(idx));
-      // Arbitrary timing
-      returnTraj->addWaypoint(idx, st->mState);
-}
-return returnTraj;  
+  std::pair <std::unique_ptr<trajectory::Interpolated>, bool> returnPair;
+  returnPair = std::make_pair(std::move(returnTraj),shorten_success); 
+  return returnPair;
```

Comment:
Nit: Why not just `return std::make_pair(std::move(returnTraj), shorten_success)`?